### PR TITLE
Small documentation improvements

### LINKED ISSUE #361: Improve docstrings
As can be seen from https://scipp.readthedocs.io/en/latest/python-reference/api.html and its subpages our doc strings generated from pybind11 are very bad.

- Is there a way to improve this?
- Are we just missing doc strings on the c++ side?
- Can we avoid polluting Python documentation with C++ type names ( see, e.g., https://scipp.readthedocs.io/en/latest/generated/scipp.Dataset.html#scipp.Dataset.__init__)?
- ...

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Fixes #361.

### CODE REVIEW

File: docs/getting-started/quick-start.ipynb [hunk 1]
```diff
@@ -0,0 +1,203 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Quick start\n",
+    "\n",
+    "This section provides a quick introduction to scipp.\n",
```
Comment (nvaytet) [inline docs/getting-started/quick-start.ipynb]:
```suggestion
    "This section provides a quick introduction to `scipp`.\n",
```

File: docs/getting-started/quick-start.ipynb [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+   "source": [
+    "x = sc.Variable(dims=[Dim.X], values=np.arange(5), unit=sc.units.m)\n",
+    "y = sc.Variable(dims=[Dim.Y], values=np.arange(4), unit=sc.units.m)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We combine the variable into a data array:"
```
Comment (nvaytet) [inline docs/getting-started/quick-start.ipynb]:
```suggestion
    "We combine the variables into a data array:"
```

File: docs/getting-started/quick-start.ipynb [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+    "sc.show(array)\n",
+    "print(array)\n",
+    "print(array.values)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Variable can have uncertainties.\n",
```
Comment (nvaytet) [inline docs/getting-started/quick-start.ipynb]:
```suggestion
    "Variables can have uncertainties.\n",
```

File: docs/getting-started/quick-start.ipynb [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+    "dataset['b'] = array\n",
+    "dataset['scalar'] = 1.23 * (sc.units.m / sc.units.s)\n",
+    "sc.show(dataset)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can slice variable, data arrays, and dataset using a dimension label and an index or a slice object like `i:j`:"
```
Comment (nvaytet) [inline docs/getting-started/quick-start.ipynb]:
```suggestion
    "We can slice variables, data arrays, and datasets using a dimension label and an index or a slice object like `i:j`:"
```

File: docs/getting-started/quick-start.ipynb [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "dataset['c'] = dataset['b'][Dim.X, 2]\n",
+    "sc.show(dataset)"
+   ]
+  },
+  {
```
Comment (nvaytet) [inline docs/getting-started/quick-start.ipynb]: Add a short comment like "Table representation and plotting is also provided" ?

File: docs/user-guide/data-structures.ipynb [hunk 6]
```diff
@@ -155,39 +158,38 @@
...
    "source": [
-    "var.shape # The sparse dimension is not part of the shape"
+    "For convenience, singular versions of the `values` and `variances` properties are provided:"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {},
+   "metadata": {
```
Comment (nvaytet) [inline docs/user-guide/data-structures.ipynb]: The downside of having docs as jupyter notebooks....

File: docs/getting-started/quick-start.ipynb (hunk 5 above)
Comment (SimonHeybrock) [reply docs/getting-started/quick-start.ipynb]: Ok, will do in the next iteration.

**Pull request merged**